### PR TITLE
remove upper bounds, fix lower bounds

### PR DESCRIPTION
**Pull request by gforsyth**
- chore: remove upper pins on default package, sqlalchemy
- chore: remove defunct patch, bump parsy lower bound

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -50,6 +51,7 @@ outputs:
         - python-dateutil >=2.8.2
         - sqlglot >=10.4.3
         - toolz >=0.11
+        - xxhash >=0.8
```
Comment (cpcloud) [inline recipe/meta.yaml]: Should these be here or in the `pooch` recipe? Ibis doesn't explicitly depend on these, we just pass them as extras.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -42,6 +42,7 @@ outputs:
         - parsy >=2
         - pooch >=1.6.0
         - poetry-dynamic-versioning >=0.18.0
+        - progress >=1.0
```
Comment (cpcloud) [inline recipe/meta.yaml]: `progress` is just the name of the extra, I believe `pooch` is using `tqdm` for progress bars.

File: recipe/meta.yaml (hunk 1 above)
Comment (gforsyth) [reply recipe/meta.yaml]: probably the pooch recipe, I'll shove them in there

File: recipe/meta.yaml (hunk 2 above)
Comment (gforsyth) [reply recipe/meta.yaml]: yeah, I got thrown because I haven't looked for an `extras` definition in a `setup.cfg` file in a while

**Review by cpcloud — APPROVED** [review]

**Pull request merged**